content/en/docs/guides/health-checking.md

```diff
@@ -0,0 +1,108 @@
+---
+title: Health Checking
+description: >-
+  Explains how gRPC servers expose a health checking service and how client can
+  be configured to automatically check the health of the server it is
+  connecting to.
+---
+
+### Overview
+
+gRPC specifies a standard service API ([health/v1]) for performing health check
+calls against gRPC servers. An implementation of this service is provided, but
+you are responsible for updating the health status of your services.
+
+On the client side you can have the client automatically communicate with the
+health services of your backends. This allows the client to avoid services that
+are considered unhealthy.
+
+### The Server Side Health Service
+
+The health check service on a gRPC server supports two modes of operation:
+
+  - Unary calls to the `Check` rpc endpoint
+    - Useful for centralized monitoring or load balancing solutions, but does
+      not scale to support a fleet of gRPC client constantly making health
+      checks
+  - Streaming health updates by using the `Watch` rpc endpoint
+    - Used by the client side health check feature in gRPC clients
+
+Enabling the health check service on your server involves the following steps:
+
+  1. Use the provided health check library to create a health check service
+  2. Add the health check service to your server.
+  3. Notify the health check library when the health of one of your services
+  changes.
+    - `NOT_SERVING` if your service cannot accept requests at the moment
+    - `SERVING` if your service is open for business
+    - If you don't care about the health of individual services, you can use
+    an empty string ("") to represent the health of your whole server.
+  4. Make sure you inform the health check library about server shutdown so
+  that it can notify all the connected clients.
+
+The exact details vary by language, see the **Language Support** section below.
+
+
+### Enabling Client Health Checking
+
+A gRPC client can be configured to perform health checks against the servers
+it connects to by modifying the [service config] of the channel. E.g. to monitor
+the health of the `foo` service you would use (in JSON format):
+
+```json
+{
+  "healthCheckConfig": {
+    "serviceName": "foo"
+  }
+}
+```
+
+Note that here you can also use an empty string if you want the channel to
+react to any service on the server becoming unhealthy.
+
+Enabling health checking changes some behavior around calling a server:
+
+  - The client will additionally call the `Watch` RPC on the health check
+  service when a connection is established
+    - If the call fails, retries will be made (with exponential backoff)
```

Comment:
append: unless the call fails with status `UNIMPLEMENTED`, in which case the health checking is disabled.